sql/core/src/test/scala/org/apache/spark/sql/errors/QueryCompilationErrorsSuite.scala

```diff
@@ -380,6 +380,24 @@ class QueryCompilationErrorsSuite
       }
     }
   }
+
+  test("INVALID_JSON_SCHEMA_MAPTYPE: " +
+    "Parse JSON rows can only contain StringType as a key type for a MapType.") {
+    withTempPath { path =>
+      val schema = StructType(
+        StructField("map", MapType(IntegerType, IntegerType, true), false) :: Nil)
+
+      checkErrorClass(
+        exception = intercept[AnalysisException] {
+          spark.read.schema(schema).json(path.getCanonicalPath)
```

Comment:
I think no need to create a temporary dir. AFAK, `json()` accepts a `Dataset`. Could you pass an empty dataset to `json()`, please.